.github/workflows/scripts/pr_workflow.py

```diff
@@ -77,22 +78,45 @@ def verify_base_config(self) -> None:
         )
 
     def set_gh_config(self) -> None:
-        gh_client: Github = Github(login_or_token=self.github_token)
-        self.repo = gh_client.get_repo(full_name_or_id=self.repo_name)
-        self.pr = self.repo.get_pull(number=self.pr_number)
+        self.gh_client: Github = Github(login_or_token=self.github_token)
+        self.repo: Repository = self.gh_client.get_repo(full_name_or_id=self.repo_name)
+        self.pr: PullRequest = self.repo.get_pull(number=self.pr_number)
 
 
 class PrLabeler(PrBaseClass):
     def __init__(self) -> None:
         super().__init__()
         self.user_login = os.getenv("GITHUB_USER_LOGIN")
         self.review_state = os.getenv("GITHUB_EVENT_REVIEW_STATE")
+        # We don't care if the body of the comment is in the discussion page or a review
         self.comment_body = os.getenv("COMMENT_BODY", "")
+        if self.comment_body == "":
+            # if it wasn't a discussion page comment, try to get a review comment, otherwise keep empty
+            self.comment_body = os.getenv("REVIEW_COMMENT_BODY", "")
         self.last_commit = list(self.pr.get_commits())[-1]
         self.last_commit_sha = self.last_commit.sha
 
+        self.verify_allowed_user()
         self.verify_labeler_config()
 
+    def get_allowed_users(self) -> list[str]:
+        org: github.Organization = self.gh_client.get_organization("opendatahub-io")
+        # slug is the team name with replaced special characters,
+        # all words to lowercase and spaces replace with a -
+        team: github.Team = org.get_team_by_slug("opendatahub-tests-contributors")
+        members: PaginatedList[github.NamedUser] = team.get_members()
+        users = [member.login for member in members]
+        # TODO: replace once bot user is part of the org and team
+        # users = ["lugi0", "rnetser", "adolfo-ab", "tarukumar", "dbasunag", "mwaykole"]
+        return users
+
+    def verify_allowed_user(self) -> None:
+        allowed_users = self.get_allowed_users()
+        if self.user_login not in allowed_users:
+            LOGGER.info(f"User {self.user_login} is not allowed for this action. Exiting.")
+            sys.exit(0)
```

Comment:
Should you exit or add a comment? Exiting would just fail the action.

Reply:
In that case how would the user know that the action was not allowed? If the workflow passed, no one is going to check for the logger message. May be you should consider adding a comment?

Reply:
why should the user know they were not allowed? I think it would be worse to see a failure every time an unallowed user tried to launch the action to be honest

Reply:
see comment above, I don't believe this is an error

Reply:
yes; error flows should exit on should exit(1). unauthorized user is an error flow

Reply:
I completely disagree, I don't see the benefit of having a failed github action in a PR if a user left a comment in the PR, especially since the action did not fail at all but simply did not proceed to perform further actions given our own permission enforcement.